.clang-tidy

```diff
@@ -0,0 +1,38 @@
+Checks: >
+    '-*,
+     bugprone-*,
+     clang-analyzer-*,
+     clang-diagnostic-*,
+     modernize-*,
+     -modernize-avoid-c-arrays,
+     -modernize-use-trailing-return-type,
+     -modernize-use-using,
+     performance-*'
+
+HeaderFilterRegex: '*parser\.(tab|lex)\.cpp\.H'
```

Comment:
Note: this filter is a whitelist for header files you want to see warnings from, not a blacklist. I think it's just reporting warnings from all header files right now, since the regex is invalid (leading asterisk).
There's currently no good way to have clang-tidy ignore specific files (see https://stackoverflow.com/q/71797349). It uses POSIX ERE syntax which doesn't support lookarounds, so I think the closest you can get is something along the lines of `[^.][^c][^p][^p]\.H$` to skip any files ending with `.cpp.H`.

Reply:
It works!  Thanks again.

Reply:
We also need to filter-in files ending with say `cqq.H`. So I think it needs to be something like `([^.][^c][^p][^p]|[^c][^p][^p]|[^p][^p]|[^p])\.H$`.

Reply:
Actually it probably should be `([^\.]...|[^c]..|[^p].|[^p])\.H$`